src/ext/java/org/opentripplanner/ext/siri/TimetableHelper.java

```diff
@@ -194,9 +194,18 @@ public static TripTimes createUpdatedTripTimes(
             realtimeDepartureTime = departureTime;
           }
 
+          boolean isCallPredictionInaccurate = Boolean.TRUE.equals(
+            recordedCall.isPredictionInaccurate()
+          );
+
           if (recordedCall.isCancellation() != null && recordedCall.isCancellation()) {
             modifiedStopTimes.get(callCounter).cancel();
             newTimes.setCancelled(callCounter);
+          } else if (isJourneyPredictionInaccurate | isCallPredictionInaccurate) {
```

Comment:
This looses the ability to mark the call as recorded

Reply:
I think this was an accepted consequence when the [StopRealTimeState](https://github.com/opentripplanner/OpenTripPlanner/blob/dev-2.x/src/main/java/org/opentripplanner/transit/model/timetable/StopRealTimeState.java)-optimization was introduced. I.e. a stop can now only have a single realtime-state.

But I agree that, ideally, it should be possible to keep multiple "realtime states" to handle some of the corner cases.